airflow/models/skipmixin.py

```diff
@@ -120,8 +120,11 @@ def skip(
             raise ValueError("dag_run is required")
 
         task_ids_list = [d.task_id for d in task_list]
-        SkipMixin._set_state_to_skipped(dag_run, task_ids_list, session)
-        session.commit()
+
+        # The following could be applied only for non-mapped tasks
```

Comment:
This should include the reasoning why doing this with an expanded task is wrong.

Reply:
Added the reasoning